airlock/policies.py

```diff
@@ -142,6 +147,17 @@ def check_can_update_file_on_request(workspace: "Workspace", relpath: UrlPath):
         )
 
 
+def check_can_withdraw_file_from_request(workspace: "Workspace", relpath: UrlPath):
```

Comment:
hmm, it probably should also do the can_edit_request check (and need a request)

Reply:
Actually, on second thoughts, I think this should stay in the permissions check rather than be a policy. It's passing the workspace so it can check the workspace file status, but the workspace file status is dependent on the user, and policies should be non-user things. I'll move it.